packages/contracts-bedrock/src/safe/TimelockGuard.sol

```diff
@@ -612,4 +646,61 @@ abstract contract TimelockGuard is IGuard {
     function signCancellation(bytes32) public {
         emit Message("This function is not meant to be called, did you mean to call cancelTransaction?");
     }
+
+    ////////////////////////////////////////////////////////////////
+    //                    Internal Functions                      //
+    ////////////////////////////////////////////////////////////////
+
+    /// @notice Internal function to clear the timelock guard configuration and cancel all pending transactions.
+    /// @dev This function does not disable the guard - it only clears the configuration state.
+    /// @param _targetSafe The Safe instance to clear the configuration for.
+    function _clearTimelockGuard(Safe _targetSafe) internal {
+        SafeState storage safeState = _safeState[_targetSafe];
+        // set the timelock delay to 0 to clear the configuration
+        safeState.timelockDelay = 0;
+
+        // Reset the cancellation threshold to 0 (unconfigured state)
+        safeState.cancellationThreshold = 0;
+
+        // Get all pending transaction hashes
+        bytes32[] memory hashes = safeState.pendingTxHashes.values();
+
+        uint256 n = hashes.length <= 100 ? hashes.length : 100;
+
+        // Cancel all pending transactions up to 100
+        // It is very unlikely that there will be more than 100 pending transactions, so we can safely limit the
+        // number of iterations to 100 in order to prevent gas limit issues.
+        // If there are more than 100 pending transactions, then we emit an event to inform the user that some
+        // transactions were not cancelled.
+        for (uint256 i = 0; i < n; i++) {
+            safeState.pendingTxHashes.remove(hashes[i]);
+            safeState.scheduledTransactions[hashes[i]].state = TransactionState.Cancelled;
+            emit TransactionCancelled(_targetSafe, hashes[i]);
+        }
+
+        if (hashes.length > 100) {
+            emit TransactionsNotCancelled(_targetSafe, hashes.length - 100);
+        }
+    }
+
+    /// @notice Internal function to disable the guard from the given Safe.
+    /// @dev This function is intended for use in contracts that extend TimelockGuard.
+    ///      It clears the timelock guard configuration and cancels all pending transactions.
+    /// @param _targetSafe The Safe instance to disable the guard from.
+    function _disableAndClearGuard(Safe _targetSafe) internal virtual {
+        // Clear the timelock guard configuration
+        _clearTimelockGuard(_targetSafe);
+
+        // Disable the guard
+        // Note that this will remove whichever guard is currently set on the Safe,
+        // even if it is not the SaferSafes guard. This is intentional, as it is possible that the guard
```

Comment:
If the current contract is not set as a Guard, it wouldn't have permissions to remove the current Guard, right?